tests/suites/test_suite_psa_crypto.function

```diff
@@ -1864,15 +2001,40 @@ void hash_compare_fail( int alg_arg, data_t *input,
 {
     psa_algorithm_t alg = alg_arg;
     psa_status_t expected_status = expected_status_arg;
+    psa_hash_operation_t operation = PSA_HASH_OPERATION_INIT;
     psa_status_t status;
 
     PSA_ASSERT( psa_crypto_init( ) );
 
+    /* Hash Compare, one-shot */
     status = psa_hash_compare( alg, input->x, input->len,
                                reference_hash->x, reference_hash->len );
     TEST_EQUAL( status, expected_status );
 
+    /* Hash Compare, multi-part */
+    status = psa_hash_setup( &operation, alg );
+    if( status == PSA_SUCCESS)
+    {
+        status = psa_hash_update( &operation, input->x, input->len );
+        if( status == PSA_SUCCESS)
+        {
+            status = psa_hash_verify( &operation, reference_hash->x,
+                                      reference_hash->len );
+            if( status != PSA_SUCCESS)
+                TEST_EQUAL( status, expected_status );
```

Comment:
This condition is not needed here, only the check. This is the last function call that need to be checked for return value which must be the same as the `expected_status`.
```suggestion
            TEST_EQUAL( status, expected_status );
```